site/project.json

```diff
@@ -1,42 +1,69 @@
 {
-  "$schema": "https://raw.githubusercontent.com/nrwl/nx/15.4.2/packages/nx/schemas/project-schema.json",
+  "$schema": "https://raw.githubusercontent.com/nrwl/nx/15.6.3/packages/nx/schemas/project-schema.json",
+  "namedInputs": {
+    "default": ["{projectRoot}/**/*"]
+  },
   "targets": {
     "start": {
-      "dependsOn": ["start-site", "playroom"],
+      "dependsOn": ["start:site", "start:playroom"],
       "executor": "nx:noop"
     },
+    "start:site": {
+      "dependsOn": ["generate"],
+      "executor": "nx:run-script",
+      "options": {
+        "script": "start:site"
+      }
+    },
+    "start:playroom": {
+      "dependsOn": ["generate"],
+      "executor": "nx:run-script",
+      "options": {
+        "script": "start:playroom"
+      }
+    },
     "build": {
-      "dependsOn": ["build-site", "build-playroom"],
+      "dependsOn": ["build:site", "build:playroom"],
       "executor": "nx:noop"
     },
-    "start-site": {
-      "dependsOn": ["generate-component-docs"],
+    "build:site": {
+      "dependsOn": ["generate"],
+      "inputs": ["default", "^default"],
+      "outputs": ["dist/(!playroom)/**/*"],
       "executor": "nx:run-script",
       "options": {
-        "script": "start-site"
+        "script": "build:site"
       }
     },
-    "build-site": {
-      "dependsOn": ["generate-component-docs"],
+    "build:playroom": {
+      "dependsOn": ["generate"],
+      "inputs": ["^default"],
+      "outputs": ["dist/playroom"],
       "executor": "nx:run-script",
       "options": {
-        "script": "build-site"
+        "script": "build:playroom"
       }
     },
-    "playroom": {
-      "dependsOn": ["generate-component-docs"],
+    "generate": {
+      "dependsOn": ["generate:component-docs", "^generate:icons"],
+      "executor": "nx:noop"
+    },
+    "generate:component-docs": {
+      "inputs": ["^default"],
+      "outputs": ["{projectRoot}/componentDocs.json"],
       "executor": "nx:run-script",
       "options": {
-        "script": "playroom"
+        "script": "generate:component-docs"
       }
     },
-    "tsc": {
-      "dependsOn": ["generate-component-docs"],
+    "lint": {},
+    "lint:tsc": {
+      "dependsOn": ["generate"],
       "executor": "nx:run-script",
       "options": {
-        "script": "tsc"
+        "script": "lint:tsc"
       }
-    }
-  },
-  "implicitDependencies": ["braid-design-system"]
+    },
+    "format": {}
```

Comment:
Confused about this change in the most recent commit. Won't this result in format scripts not running?

Reply:
I encourage you to verify it locally.

This is a not-so-well documented feature that is briefly mentioned in passing [in the docs](https://nx.dev/reference/nx-json#target-defaults). If you look at Nx's own configuration you'll notice the same pattern.

Top-level `nx.json`: https://github.com/nrwl/nx/blob/master/nx.json#L88
Project config: https://github.com/nrwl/nx/blob/master/packages/next/project.json#L59